
## 控制器编号: 0 for 串级PID, 1 for PID, 2 for UDE, 3 for passivity, 4 for NE
px4_pos_controller:
  controller_number: 1

## 位置环参数 - cascade pid
## 目前参数为PX4默认参数
Pos_cascade_pid:
  Kp_xy : 0.8
  Kp_z : 1.0
  Kp_vxvy : 0.2
  Kp_vz : 0.4
  Ki_vxvy : 0.02
  Ki_vz : 0.15
  Kd_vxvy : 0.016
  Kd_vz : 0.0
  ##悬停油门，取决于飞机重量
  Hover_throttle : 0.42
  MPC_VELD_LP: 5.0

## 限制参数 
Limit:
  tilt_max: 20.0
  XY_VEL_MAX : 1.0
  Z_VEL_MAX : 1.0
  THR_MIN: 0.1
  THR_MAX: 0.9
  pxy_error_max: 10.0
  vxy_error_max: 10.0
  pz_error_max: 10.0
  vz_error_max: 10.0
  pxy_int_max : 10.0
  pz_int_max : 10.0
  int_start_error: 10.0

## 飞机参数
Quad:
  mass: 0.9

# 位置环参数 - normal pid
Pos_pid:
  Kp_xy : 2.5
  Kp_z : 2.0
  Ki_xy : 0.5
  Ki_z : 0.5
  Kd_xy : 3.0
  Kd_z : 3.0

# 位置环参数 for ude
Pos_ude:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 2.0
  T_ude_z : 2.0


# 位置环参数 for passivity
Pos_passivity:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 1.0
  T_ude_z : 1.0
  T_ps: 0.2

# 位置环参数 for ne
Pos_ne:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 1.0
  T_ude_z : 1.0
  T_ne : 0.1





